redbot/cogs/warnings/warnings.py

```diff
@@ -342,45 +342,57 @@ def __init__(self, bot: Red):
 
     @commands.command()
     @commands.guild_only()
-    async def warnings(
-        self, ctx: commands.Context, user: Optional[Union[discord.Member, int]] = None
-    ):
-        """List the warnings for the specified user.
-
-        Omit `<user>` to see your own warnings.
-
-        Note that showing warnings for users other than yourself requires
-        appropriate permissions.
-        """
-        if user is None:
-            user = ctx.author
-        else:
-            if not await is_admin_or_superior(self.bot, ctx.author):
-                return await ctx.send(
-                    warning(_("You are not allowed to check warnings for other users!"))
+    @checks.admin_or_permissions(ban_members=True)
```

Comment:
This should just be `checks.admin()` matching the prior behavior handled inside the command.